### PR TITLE
Wizard recipe: Clay-v0.1.2

### PR DESCRIPTION
**Pull request by jkroso**
This pull request contains a new build recipe I built using the BinaryBuilder.jl wizard:

* Package name: Clay
* Version: v0.1.2

@staticfloat please review and merge.

### CODE REVIEW

**Comment by jkroso** [comment]
The linux platforms I disabled are because of the following error for those wondering:
```
/opt/i686-linux-gnu/bin/i686-linux-gnu-ld: /tmp/clay-beb7b4.o: unrecognized relocation (0x2b) in section `.text'
/opt/i686-linux-gnu/bin/i686-linux-gnu-ld: final link failed: Bad value
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

File: C/Clay/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+if [ "$PLATFORM_NAME" = "Darwin" ]; then
+    mkdir -p ${prefix}/lib
+    clang -shared ${WORKSPACE}/srcdir/clay/clay.c -fPIC -o ${prefix}/lib/libclay.dylib
+elif [ "$PLATFORM_NAME" = "Linux" ] || [ "$PLATFORM_NAME" = "FreeBSD" ]; then
+    mkdir -p ${prefix}/lib
+    clang -shared ${WORKSPACE}/srcdir/clay/clay.c -fPIC -o ${prefix}/lib/libclay.so
+else
+    mkdir -p ${prefix}/bin
+    clang -shared ${WORKSPACE}/srcdir/clay/clay.c -fPIC -o ${prefix}/bin/libclay.dll
+fi
```
Comment (giordano) [inline C/Clay/build_tarballs.jl:28]: We have many [automatic environment variables](https://docs.binarybuilder.org/stable/build_tips/#Automatic-environment-variables) to simplify this:
```suggestion
mkdir -p "${libdir}"
cc -shared -fPIC -o "${libdir}/libclay.${dlext}" clay.c
```

File: C/Clay/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    # Platform("i686", "linux"),
+    # Platform("x86_64", "linux"),
+    Platform("i686", "Windows"),
+    Platform("x86_64", "Windows"),
+    Platform("x86_64", "FreeBSD"),
+    Platform("aarch64", "FreeBSD"),
+    Platform("aarch64", "macos";),
+    Platform("x86_64", "macos";),
+]
+
```
Comment (giordano) [inline C/Clay/build_tarballs.jl:48]:
```suggestion
platforms = supported_platforms()
```

Comment (jkroso) [reply C/Clay/build_tarballs.jl:39]: But how would I disable to 2 that I commented out?

File: C/Clay/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+# The products that we will ensure are always built
+products = [
+    LibraryProduct("libclay", :libclay)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = Dependency[]
+
+# Build the tarballs, and possibly a `build.jl` as well.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6")
```
Comment (giordano) [inline C/Clay/build_tarballs.jl:50]:
```suggestion
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; julia_compat="1.6", preferred_gcc_version=v"6")
```

**Comment by giordano** [comment]
Thanks for your contribution!

**Comment by jkroso** [comment]
Thanks for your help, it’s heaps better now!

**Pull request merged**